src/power_grid_model/utils.py

```diff
@@ -301,3 +304,62 @@ def _compatibility_deprecated_import_json_data(json_file: Path, data_type: Datas
         raise PowerGridSerializationError("An internal error occured during deserialization")
 
     return result
+
+
+def self_test():
+    """
+    Perform a self-test of the Power Grid Model functionality to check if the installation was successful
+    and there are no build errors, segmentation violations, undefined symbols, etc.
+
+    This function is designed to validate the basic functionality of data serialization,
+    model instantiation, power flow calculation, and serialization of results using the
+    Power Grid Model library.
```

Comment:
the line sentence of the documentation should be a short single-line description (no details). The details about what it does should be included in the rest of the docstring

```suggestion
    """
    Perform a self-test of the Power Grid Model functionality.

    Tests whether the installation was successful and there are no build errors,
    segmentation violations, undefined symbols, etc.

    This function is designed to validate the basic functionality of data serialization,
    model instantiation, power flow calculation, and serialization of results using the
    Power Grid Model library.
```